### PR TITLE
Added parts to run js tests in drone

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Starting to add more jobs into drone.

### CODE REVIEW

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
- [ ] remoce the js step from Jenkins file

File: .drone.yml [hunk 1]
```diff
@@ -98,6 +98,21 @@ pipeline:
       matrix:
         TEST_SUITE: coverage
 
+  javascript:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    group: test
+    environment:
+      - PHP_VERSION=${PHP_VERSION}
```
Comment (DeepDiver1975) [inline .drone.yml]: both environments are not used in test-javascript.sh - right? we can drop them from my understanding

**Comment by SergioBertolinSG** [comment]
Failed test in javascript:

```javascript
PhantomJS 2.1.1 (Linux 0.0.0) OC.Share.ShareDialogView autocompletion of users is sorted naturally FAILED
	Expected $[1].label = 'Petra' to equal 'peter B.'.
	Expected $[1].value.shareWith = 'Petra' to equal 'peter B.'.
	Expected $[2].label = 'peter B.' to equal 'Petra'.
	Expected $[2].value.shareWith = 'peter B.' to equal 'Petra'.
	core/js/tests/specs/sharedialogviewSpec.js:282:47
```


Some steps to debug the javascript failed test locally:

1. Install drone cli http://docs.drone.io/cli-installation
2. Clone core's master
3. Checkout my branch drone-enabling-js-tests
4. Change core's .drone.yml with my file https://gist.github.com/SergioBertolinSG/8394e95c9cb678f745b0348674daa6de
5. Run it this way: PHP_VERSION=7.1 drone exec

cc @PVince81

**Comment by SergioBertolinSG** [comment]
Some error in jenkins:

ERROR: No test report files were found. Configuration error?
Finished: FAILURE

**Comment by SergioBertolinSG** [comment]
PHP 7.2 is not yet supported AFAIK.
MariaDB is failing because:

```sql
Error while trying to create admin user: An exception occurred while executing 'ALTER TABLE oc_calendarsubscriptions CHANGE `lastmodified` lastmodified INT UNSIGNED DEFAULT NULL NOT NULL':

SQLSTATE[42000]: Syntax error or access violation: 1067 Invalid default value for 'lastmodified'
 ->
```

Do we need to have both MariaDB and MySQL ?

**Comment by SergioBertolinSG** [comment]
OK, so I have skip the failing test: https://github.com/owncloud/core/pull/29299/files#diff-e18e03b16d13b8606ea5c1cfe595a23fR232

I'll open an issue in core after this is merged to fix that failing test. cc @PVince81

File: .drone.yml [hunk 2]
```diff
@@ -98,6 +98,18 @@ pipeline:
...
 
+  javascript:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    group: test
+    commands:
+      - ./tests/drone/test-javascript.sh
+    when:
+      event: [push, pull_request]
+      branch: [master, stable10, stable9.1]
```
Comment (DeepDiver1975) [inline .drone.yml]: remove branch here and rebase with master - see https://github.com/owncloud/core/blob/master/.drone.yml#L5

thx

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by DeepDiver1975** [comment]
@SergioBertolinSG jenkins still hates this ..... please adopt. THX

**Comment by PVince81** [comment]
> jenkins still hates this

Hahaha, Jenkins doesn't want to be obsoleted

**Comment by PVince81** [comment]
Uh oh:
```
Test\Files\ViewTest::testCopyBetweenStorageNoCross
mkdir(): No space left on device
```

File: .drone.yml [hunk 3]
```diff
@@ -91,6 +91,17 @@ pipeline:
...
         TEST_SUITE: coverage
 
+  javascript:
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    group: test
+    commands:
+      - ./tests/drone/test-javascript.sh
+    when:
+      event: [push, pull_request]
```
Comment (tboerger) [inline .drone.yml]: Drop that, push and pull request is the default behavior

**Comment by DeepDiver1975** [comment]
squash please and we are ready to go - THX

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by DeepDiver1975** [comment]
@SergioBertolinSG please backport to stable10 - for 9.1 please add this commit to #29236 - THX

**Comment by SergioBertolinSG** [comment]
Backports
stable10: https://github.com/owncloud/core/pull/29387
stable9.1: Included in https://github.com/owncloud/core/pull/29236

**Pull request merged**